controllers/collection.lua

```diff
@@ -87,53 +78,106 @@ CollectionController.GET.user_collections = function (self)
+        local published = params.published ~= nil and params.published == true
+        local published_at = (published and collection.published_at) or
+            current_time_or_nil(published)
+        local shared = params.shared ~= nil and params.shared == true
+        local shared_at = (shared and collection.shared_at) or
+             current_time_or_nil(shared)
+
+        collection:update({
+            name = params.name or collection.name,
+            slug = params.name and util.slugify(params.name) or collection.slug,
+            description = params.description or collection.description,
+            published = published,
+            published_at = published_at,
+            shared = shared,
+            shared_at = shared_at,
+            thumbnail_id = params.thumbnail_id or collection.thumbnail_id
+        })
+
+        return jsonResponse(collection)
+    end
 
+    -- Must assert name before generating a slug.
+    validate.assert_valid(params, { { 'name', exists = true } })
     return jsonResponse(assert_error(Collections:create({
-        name = self.params.name,
-        slug = util.slugify(self.params.name),
-        creator_id = self.request_user.id,
-        description = self.params.description,
-        published = self.params.published,
-        published_at = current_time_or_nil(self.params.published),
-        shared = self.params.shared,
-        shared_at = current_time_or_nil(self.params.shared),
-        thumbnail_id = self.params.thumbnail_id
+        name = params.name,
+        slug = util.slugify(params.name),
+        creator_id = queried_user.id,
+        description = params.description,
+        published = params.published  == true,
+        published_at = current_time_or_nil(params.published),
+        shared = params.shared  == true,
+        shared_at = current_time_or_nil(params.shared),
+        thumbnail_id = params.thumbnail_id
     })))
+end)
+
+CollectionController.DELETE.collections = function (self)
+    -- DELETE /users/:username/collections/:collection_slug
+    -- Description: Delete a particular collection.
+end
+
+CollectionController.GET.collection_memberships = function (self)
+    -- GET /users/:username/collections/:collection_slug/projects(/:project_id)
+    -- Description: Get a paginated list of all projects in a collection.
+    -- Parameters:  username, collection_slug
+    local collection = assert_collection_exists(self)
+    if self.params.project_id then
+        return jsonResponse(CollectionMemberships:find(collection.id,
+                                                       self.params.project_id))
+    end
+
+    -- TODO: Not sure how to pass a pagesize to this. :/
+    -- Note: May need to re-write this as a method w/o using the `relations`
+    local projects = collection:get_projects()
+    return jsonResponse(projects:get_page(self.params.page or 1))
 end
 
 CollectionController.POST.collection_memberships = function (self)
```

Comment:
Again, for Controller methods, I'd use names that convey what the function does, not what the URL looks like. I'd call this one something like `add_project`.